test/Microsoft.Windows.CsWin32.Tests/Microsoft.Windows.CsWin32.Tests.csproj

```diff
@@ -1,15 +1,27 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\src\Microsoft.Windows.CsWin32\build\Microsoft.Windows.CsWin32.props" />
 
   <PropertyGroup>
     <TargetFrameworks>net5.0;netcoreapp3.1;net472</TargetFrameworks>
     <RootNamespace />
   </PropertyGroup>
 
+  <PropertyGroup>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <CompilerGeneratedFilesOutputPath>$(MSBuildProjectDirectory)\.generated\$(TargetFramework)</CompilerGeneratedFilesOutputPath>
```

Comment:
FWIW: the source generator may customize its output by more dimensions than just TFM. And the TFMs this test project happens to build with doesn't necessarily represent all possibilities either. 
Tying the test runtime itself to the source generation feels a bit like an inappropriate coupling, as a single build of the test assembly should be able to test all possible outputs of the SG as run by that runtime. For example, a source generator running on `net472` should be able to generate source in a project targeting `net6.0`. This test project today can do that, but this generated code thing that you're adding wouldn't.

Reply:
I wonder if we should instead have tests have a switchable read/write mode, where they emit the generated files to their source directory when run under "update LKG" mode, and otherwise they read from those same locations.
That might help avoid the risk you bring up of cross-test contamination from NativeMethods.txt changes if each test still gets to manage its own LKG files.

Reply:
> In the future, satellite projects could be used to generate test inputs under fixed configurations with similar results, or the test implementation could be updated to write expected outputs back to disk when the tests are executed.

Sounds like you had similar ideas.